### ISSUE TITLE
Support linting from stdin / in-memory source

### ISSUE THREAD

**Issue by lekemula** [mensfeld/yard-lint#173]
## Motivation

`yard-lint` currently reads source from disk only. This works fine for CI and pre-commit use, but it limits LSP/editor integration: an editor buffer with unsaved edits can't be linted until the user hits save.

Concretely, [solargraph-yard-lint](https://github.com/lekemula/solargraph-yard-lint) surfaces yard-lint offenses as Solargraph LSP diagnostics, but has to gate on `File.file?(filename)` and skip linting any buffer whose contents diverge from disk ([source](https://github.com/lekemula/solargraph-yard-lint/blob/main/lib/solargraph/yard_lint/diagnostic.rb)). The result is the editor stops surfacing yard-lint warnings the moment the user starts typing — which is exactly when linting feedback is most useful. Other reporters in the same Solargraph pipeline (`rubocop`, `typecheck`) lint the in-memory buffer and don't have this gap.

## Prior art: RuboCop

RuboCop has supported this for years via `--stdin`. Solargraph's RuboCop reporter passes the buffer contents directly:

```ruby
# solargraph/lib/solargraph/diagnostics/rubocop_helpers.rb
def generate_options(filename, code)
  args = ['-f', 'j', '--force-exclusion', filename]
  options, paths = RuboCop::Options.new.parse(args)
  options[:stdin] = code
  [options, paths]
end

# solargraph/lib/solargraph/diagnostics/rubocop.rb
runner = RuboCop::Runner.new(options, store)
runner.run(paths)
```

`filename` is still passed so config resolution, `Include`/`Exclude` matching, and offense locations all work — only the source bytes come from memory.

CLI form:

```sh
echo "code" | rubocop --stdin path/to/file.rb
```

## Proposal

Mirror that shape in yard-lint:

```ruby
Yard::Lint.run(path: 'foo.rb', source: buffer_contents, progress: false)
```

The `path` keeps config/exclusion/reporting behavior intact; the source bytes come from the caller.

## Context

Discussed briefly in https://github.com/mensfeld/yard-lint/issues/17#issuecomment-4654100706 — opening a dedicated issue so it doesn't get buried in the LSP thread. Happy to take a stab at the PR if you confirm the API shape you'd prefer.

**Comment by mensfeld** [comment]
@lekemula in master. Can you check that it works in an expected way?

**Comment by lekemula** [comment]
@mensfeld works like a charm! 🚀  https://github.com/lekemula/solargraph-yard-lint/pull/1

**Comment by mensfeld** [comment]
Amazing! I should ship some good fixes in the upcoming week.
